plugins/module_utils/network/ios/facts/vlans/vlans.py

```diff
@@ -44,161 +45,67 @@ def __init__(self, module, subspec="config", options="options"):
 
         self.generated_spec = utils.generate_dict(facts_argument_spec)
 
-    def get_vlans_data(self, connection, configuration):
+    def get_vlans_data(self, connection):
         """Checks device is L2/L3 and returns
         facts gracefully. Does not fail module.
         """
-        if configuration:
-            cmd = "show running-config | sec ^vlan configuration .+"
-        else:
-            cmd = "show vlan"
         check_os_type = connection.get_device_info()
         if check_os_type.get("network_os_type") == "L3":
             return ""
-        return connection.get(cmd)
+        return connection.get("show vlan")
+
+    def get_vlan_conf_data(self, connection):
+        return connection.get("show running-config | sec ^vlan configuration .+")
+
+    def populate_vlans_config_facts(self, connection, data=None):
+        """Process config for Vlans Configurations
 
-    def populate_facts(self, connection, ansible_facts, data=None):
-        """Populate the facts for vlans
         :param connection: the device connection
         :param ansible_facts: Facts dictionary
         :param data: previously collected conf
+
         :rtype: dictionary
         :returns: facts
         """
-        configuration = self._module.params["configuration"]
-        objs = []
 
-        if not data:
-            data = self.get_vlans_data(connection, configuration)
-        if not configuration:
-            objs = self.parse_vlan(data)
-        else:
-            objs = self.parse_vlan_config(data)
+        if not data or "configuration" not in data:
```

Comment:
I'm trying to understand why the second condition is required. IIRC, this check if for `state: parsed`, if so, both the configuration stanzas should be given one after the other (with no empty line) and should be treated like a singular entity. Shouldn't the first condition suffice?